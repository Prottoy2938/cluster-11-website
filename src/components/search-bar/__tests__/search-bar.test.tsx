import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import SearchBar from "../search-bar";

afterEach(cleanup);

test("should take and match snapshot", () => {
  const { asFragment } = render(<SearchBar />);
  expect(asFragment(<SearchBar />)).toMatchSnapshot();
});

//this test will fail unless you give the search id and key to run the test. I can't make react-testing-library to access my my .env file.
//If you can't access your process.env data then you can do a workaround. Pass the value of `searchID` and `searchKey` when you run `npm run test` from the command line.
//like this
//$ searchID="you search key" searchKey="your search id" npm run test

//TODO:   UNUSED TEST, HAVE TO EDIT THIS
// test("Shows search suggestions", async () => {
//   const { getByTestId } = render(<SearchBar />);
//   const input = getByTestId("search-field");
//   fireEvent.change(input, { target: { value: "chic" } });
//   const suggestions = await waitForElement(() => getByTestId("1"));
//   expect(suggestions).toHaveTextContent("Chicken Pho");
// });

//theres something wrong with this test, I can't figure it out
test("Remove search query onFocus and back space key", () => {
  const { getByTestId } = render(<SearchBar />);
  const input = getByTestId("search-field");
  fireEvent.focus(input);
  fireEvent.keyDown(input, { key: "Backspace", code: "Backspace" });
  expect(input).toHaveTextContent("");
});
