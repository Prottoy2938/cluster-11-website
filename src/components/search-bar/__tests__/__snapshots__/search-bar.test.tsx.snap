// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="searchContainer"
    >
      <input
        autocomplete="off"
        class="searchField"
        data-testid="search-field"
        id="search-field"
        spellcheck="false"
        type="text"
        value=""
      />
      <button
        aria-label="search"
        class="MuiButtonBase-root MuiIconButton-root searchIconContainer MuiIconButton-colorPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="suggestionContainer"
    />
  </div>
</DocumentFragment>
`;
